sunbird_mcp/src/server.py

```
@@ -0,0 +1,384 @@
+        # Validate the returned content_ids
+        if not content_ids:
+            return json.dumps({"error": "No content IDs found for the given content_id"}, ensure_ascii=False)
+            
+        if not isinstance(content_ids, list):
+            return json.dumps({"error": "Unexpected response format: content_ids is not a list"}, ensure_ascii=False)
+
+        
+        artifact_urls = []
+
+        # Create a semaphore to limit concurrent requests
+        semaphore = asyncio.Semaphore(20)  # Limit to 20 concurrent requests
```

Ruff (0.11.9)

Local variable `semaphore` is assigned to but never used

Remove assignment to unused variable `semaphore`

(F841)